### PR TITLE
Revise main Getting started tutorial for clarity

### PR DESCRIPTION
**Pull request by kev-odin**
No code changes. Only documentation revision for the getting started tutorial. I wanted to learn Wagtail, so I started with the tutorial and made changes along the way when I got confused.

Definitely open to any feedback or improvements that I can make :) 

Changes:
1. Addition of learning objectives describing what the tutorial will accomplish with the learner.
2. Added base prerequisite knowledge and the description of the intended audience of the tutorial.
3. Fixed up typos and a couple of inconsistent language occurrences.
4. Rewrote sections that were lengthy and used numbered lists for complex steps
5. Simple explanations for admin users, database migrations, SQLite, and Wagtail's QuerySet modifiers.
6. Added hierarchical tree structure visuals.
7. Improved call to action at the end of tutorial.
8. 100% more jazz hands (when I was relieved something worked)

Stayed the same:
1. General tutorial layout, I found it adequate to get me started
2. Tutorial images, I saw no reason to change, since I think the interfaces are the same.
3. Odd references like `(virtual_environment_creation)=` and `(tutorial_categories)=` which I do not see a reference anywhere in the document?

Tested on Ubuntu 22.04 LTS and Windows 10.

I ran it through a word processor to catch spelling mistakes and generated a working local version using Sphinx.

_Please check the following:_

-   [x] Do the tests still pass?[^1]
-   [x] Does the code comply with the style guide?
    -   [ ] Run `make lint` from the Wagtail root.
-   [ ] For Python changes: Have you added tests to cover the new/fixed behaviour?
-   [ ] For front-end changes: Did you test on all of Wagtail’s supported environments?[^2]
    -   [ ] **Please list the exact browser and operating system versions you tested**:
    -   [ ] **Please list which assistive technologies [^3] you tested**:
-   [ ] For new features: Has the documentation been updated accordingly?

### CODE REVIEW

**Review by lb- — CHANGES_REQUESTED** [review]
Thanks @kev-odin  there is a lot to take in here, it is very likely that this PR will take a while to get reviewed fully in its current state.

Is there any way you can maybe break this up into some smaller quick wins (e.g. PRs that just add the right `docson` / Windows versions of the code snippets?

Maybe another core team member will have the time to review in full and have a different take on this but just trying to help you get to a point where some things can be merged and the more 

We really appreciate the effort but just want to ensure that this does not sit in limbo with over 650 lines of changes touching almost every part of this critical path page.

From what I can see, there are broadly these kinds of changes;

1. Clean up /refinement of bash snippets - mostly great, thank you.
2. Update to `python3` globally - this is a controversial one, we do not do this much throughout the rest of the documentation and this also has had lots of back and forth on other PRs, not sure it's worth tackling here so maybe keep it all as `python`
3. Learning objectives/intro/pre-reqs - this is a good addition and could easily be a standalone PR without too much conflicts with other changes.
4. Adding full stops / grammar changes  - nice to have, but can be a standalone PR.
5. Minor typos (e.g. the spelling of browswer vs browser - pull these to a standalone PR if possible, much easier to merge in quickly.
6. Small beginner tips (adding to .gitignore note, folder structure tree, )
7. Title capitalisation changes - while not explicitly called out, we do not really use all upper case titles, we just title case the proper nouns, please revert those changes as they just add noise to the PR. Maybe it's worth a new PR (which is welcome) to be explicit about title casing in https://docs.wagtail.org/en/stable/contributing/documentation_guidelines.html  . You appear to have a mix of APA Title Case (more suitable for book titles, newspaper articles), but the current docs have sentence casing for titles where only proper nouns and Models are capitalised.
8. Rewording for rewording sake, this adds noise to the PR and without a clear readability improvement it just feels like churn in the code because someone put the text through grammarly or chat gpt. Sometimes wording changes can help but just want to ensure the changes stay purposeful and focused. Adjusting the wording of a statement can sometimes be beneficial, but it is important to ensure that any changes made are intentional and relevant to the topic at hand. :) 
9. Tone changes - while appreciated, maybe we can keep the tone more neutral and not too frivolous, but this is just my opinion.
10. Where to next - could be a good standalone PR
11. Finally. We get to a bit of a wide range of changes that either flesh out content or completely add/reword things. It would be great to give feedback on these specifically but with all the noise of above it's hard to know where to start.

## General feedback

* A reminder that spelling must be British English (e.g. recognised not recognized, organised not organized)
* Please have a full read of https://docs.wagtail.org/en/stable/contributing/documentation_guidelines.html
* General coding guidelines https://docs.wagtail.org/en/stable/contributing/general_guidelines.html (has notes about some documentation items also)
* Maybe avoid too much semi-colon usage, try to reword to a sentence
* No need to add the changelog entry, core team will add this stuff when merged, this way it does not conflict with other changes to those files that come in as the PR is open.

**Review by thibaudcolas — COMMENTED** [review]
@kev-odin thanks for giving this a go. In the future when attempting large changes like this, I’d strongly recommend you consult with us ahead of time. 

I see two specific problems with the way you’ve gone about this as-is:

- We’ve already been planning to [enhance this tutorial and write more](https://wagtail.org/blog/google-season-of-docs-creating-wagtail-developer-onboarding-tutorials/), starting in May/June. That doesn’t take anything away from your proposed changes, but means some of this work might just get re-done anyway if we decide to take a different direction.
- You do lots of different types of changes in one go, which means it’s harder for us to integrate the changes that are pretty clearly beneficial.

Overall I think your contributions look solid, and I think you were right to avoid code changes and screenshot changes to keep this easier to integrate – but in the future do share specific points of confusion and general suggested improvements with us first, so we can guide your contributions based on our plans.

**Comment by lb-** [comment]
One more helpful resource from our docs for your first contribution - https://docs.wagtail.org/en/stable/contributing/first_contribution_guide.html

It includes some of the general feedback already given but hopefully helps you get to a first few PRs that can be easily reviewed and merged in.

**Comment by kev-odin** [comment]
@lb- @thibaudcolas 
 
Thank you both so much for the feedback! I can definitely split these edits into smaller reviewable chunks. Those resources that you listed are incredibly helpful, I am a newbie with contributing to open source so I really appreciate the guidance.

I will work on getting these suggested changes in the coming week. Spending most of my free time in Hyrule this weekend :) 

To be clear, I should open separate PRs for each edits to make it easier for the core team to review? Does that mean we should keep this PR open and reference the smaller PRs to this specific thread?

I do not mind if the tutorial goes in a different direction, this exercise helped me solidify the concepts after writing about it and hopefully this is useful to others in their journey. The most confusing portion of the tutorial was the QuerySet modifiers, which is why I created a simple tree visual for the reader. Also it took me some time to figure out why Parents and Children pages were referenced before the Tree hierarchy explanation.

Cheers!

File: docs/getting_started/tutorial.md [hunk 1]
```diff
@@ -155,30 +234,46 @@ class HomePage(Page):
...
-capabilities and the layout of the editing interface. When you add fields to `content_panels`, it enables them to be edited on the Wagtail interface. [More on creating Page models](../topics/pages).
+In this code snippet above, `body` is defined as `RichTextField`, a special Wagtail field.
+When `blank=True`, this field is not required and can be empty.
 
-Run:
+You can use any of the [Django core fields](https://docs.djangoproject.com/en/stable/ref/models/fields). Notice that `content_panels` define the
+capabilities and the layout of the editing interface. When you add fields to `content_panels`,
+it enables these fields to be edited in the Wagtail interface. [More on creating `Page` models](../topics/pages).
+
+Before we can see the changes, we will need to perform a database migration. Simply explained, a database migration refers to the process of moving data from one database to another. In this case, our initial empty database to a new database with a `HomePage` model.
```
Comment (gasman) [inline docs/getting_started/tutorial.md:244]: This isn't really correct - in this case "database migration" refers to changing the structure of the database while keeping its data intact, not moving data between databases.

**Comment by lb-** [comment]
@kev-odin in regards to your question about the PR. It's up to you what you want to do. You can close this and open discrete PRs or change this PR to reflect your first desired change based on feedback. 

Maybe in the meantime you can change this to a draft until ready for another review.

**Review by thibaudcolas — APPROVED** [review]
After consulting with @vossisboss and @activus-d, we’ve decided to take some time to break up this pull request ourselves and at least merge the changes that everyone is in agreement are a net improvement.

That’s still a fair amount of changes – thank you @kev-odin for making this happen!

I have rebased the pull request accordingly, reverting all of the changes that weren’t clear improvements over the existing content. 

For ref, here is the previous version of the changes: https://github.com/wagtail/wagtail/commit/c31af2759542944212ed5c2471f6f71de355eb45.

**Pull request merged**